packages/storage-plus/src/de.rs

```diff
@@ -0,0 +1,171 @@
+        Ok(value.to_vec())
+    }
+}
+
+impl KeyDeserialize for &Vec<u8> {
+    type Output = Vec<u8>;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        <Vec<u8>>::from_slice(value)
+    }
+}
+
+impl KeyDeserialize for [u8] {
+    type Output = Vec<u8>;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        <Vec<u8>>::from_slice(value)
+    }
+}
+
+impl KeyDeserialize for &[u8] {
+    type Output = Vec<u8>;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        <Vec<u8>>::from_slice(value)
+    }
+}
+
+impl KeyDeserialize for String {
+    type Output = String;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        String::from_utf8(value.to_vec())
+            // FIXME: Add and use StdError utf-8 error From helper
+            .map_err(|err| StdError::generic_err(err.to_string()))
+    }
+}
+
+impl KeyDeserialize for &String {
+    type Output = String;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        String::from_slice(value)
+    }
+}
+
+impl KeyDeserialize for str {
+    type Output = String;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        String::from_slice(value)
+    }
+}
+
+impl KeyDeserialize for &str {
+    type Output = String;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        String::from_slice(value)
+    }
+}
+
+impl KeyDeserialize for Addr {
+    type Output = String;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        String::from_slice(value)
+    }
+}
+
+impl KeyDeserialize for &Addr {
+    type Output = String;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        String::from_slice(value)
+    }
+}
+
+macro_rules! integer_de {
+    (for $($t:ty),+) => {
+        $(impl KeyDeserialize for IntKey<$t> {
+            type Output = $t;
+
+            fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+                Ok(<$t>::from_be_bytes(value.try_into()
+                    // FIXME: Add and use StdError try-from error From helper
+                    .map_err(|err: TryFromSliceError| StdError::generic_err(err.to_string()))?))
+            }
+        })*
+    }
+}
+
+integer_de!(for i8, u8, i16, u16, i32, u32, i64, u64, i128, u128);
+
+impl KeyDeserialize for TimestampKey {
+    type Output = u64;
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        Ok(<u64>::from_be_bytes(
+            value
+                .try_into()
+                // FIXME: Add and use StdError try-from error From helper
+                .map_err(|err: TryFromSliceError| StdError::generic_err(err.to_string()))?,
+        ))
+    }
+}
+
+impl<T: KeyDeserialize, U: KeyDeserialize> KeyDeserialize for (T, U) {
+    type Output = (T::Output, U::Output);
+
+    fn from_slice(value: &[u8]) -> StdResult<Self::Output> {
+        let (len, data) = value.split_at(2);
+        let t_len = u16::from_be_bytes(
```

Comment:
This method `{u16,u64,...}::from_be_bytes(....)` is called 5 places in this code. It would be nice to pull it out into a helper method.

Reply:
Hmm, not possible as is because the integer types impl of `from_be_bytes` is not associated to a trait. So, I need to be explicit when calling these types. Macros do it, but fns don't, as I need to bound the types.

A suggestion in SO is to use the `Num` trait from `num` crate, but I'm hesitant to introduce a new dep just for a trait.

I could also implement my own marker trait, and add a `from_be_bytes` in there like in `Endian`, but this also looks overkill to me. I would just leave these calls as they are. When we implement the error helper in cosmwasm-std, they will look much nicer.